### ISSUE TITLE
Syncing fails with 'A required privilege is not held by the client' on Windows with symlinks

### ISSUE THREAD

**Issue by obambrough** [syncthing/syncthing#4035]
After upgrading to Syncthing v0.14.24 I'm no longer able to finish syncing between two Windows machines (it won't ignore symbolic links anymore which it isn't able to sync.)

Machine A has a 'send only' folder that is synchronized with Machine B. Some files on machine A are symbolic links. Previously I'd turned off symlinks under the advanced configuration options so the symbolic links would be ignored. The option to ignore symbolic links has been removed in v0.14.24 and now it won't ignore them and it can't synchronize them either.

### Version Information

Machine A:
Syncthing Version: v0.14.24
OS Version: Windows 10

Machine B:
Syncthing Version: v0.14.24
OS Version: Windows 10

Probably unrelated but I should note I use tray icon Synctrayzor (at latest, v1.1.14) on both machines which starts Syncthing.

**Comment by AudriusButkevicius** [comment]
As the issue template explains, please use the [forum](https://forum.syncthing.net/) for support.
Please use the search on the forum and the closed issue tracker for simillar issues before starting a new thread. 
If this issue is later proven to be a bug, we will reopen this issue or open a new one.

**Comment by calmh** [comment]
This does actually sound like a real bug though to me. Symlinks between two windows boxes should certainly not cause permission errors. I'd like for someone to see if this is reproducible or not.

**Comment by obambrough** [comment]
Just some more info. I fired up an older Windows VM that was at Syncthing v0.14.6 and have documented with screenshots what happens under Windows between a new version of Syncthing v0.14.24 and the older 0.14.6 version, then upgrading the older machine to the latest version of Syncthing.

The screenshots are here: https://1drv.ms/f/s!AqWtxefKp-f5nla2eMKXHG6PCo9x
There's one screenshot taken from each machine and given a matching number (so 1 - Fractal to Scratch, matches 1 - Scratch to Fractal.)

Images no. 1) Two folders had been previously shared between them and they must have contained a symlink, Fractal (at 0.14.24) warns it won't sync symlinks, and Scratch (at 0.14.6) won't finish syncing.

Images no 2) I share a new folder (TestNoSymlinks) and stop sharing the older folders that must contain symlinks - syncing completes for both machines.

Images no 3) I share a new folder (TestSymlinks) and create a symlink on Scratch (at 0.14.6.) Everything syncs fine (actual symlink isn't synced between the machines, but no errors or syncing issues are present.)

Images no 4) I create a symlink on Fractal (at 0.14.24) Everything appears fine on Scratch but Fractal is stuck at 99%

Images no 5) I upgrade Scratch to 0.14.24 now both machines are stuck at 99% syncing and show the TestSymlink folder as 'Out of Sync'. Images no 5-1 show the items that failed (they're both symlinks.)

**Comment by calmh** [comment]
How are the symlinks created?

**Comment by obambrough** [comment]
On Scratch the symlink was created via command line

    mklink "C:\Users\xxx\TestSymlinks\subfolder-from-scratch\test-symlink2.plgx" "C:\Users\xxx\TestSymlinks\SourceForgeUpdateChecker.plgx"

On Fractal the symlink was created via a shell plugin that allows me to drag a file to another folder and drop either a Hardlink or a Symbolic Link - in this case I dropped a Symbolic Link.

**Comment by calmh** [comment]
Confirmed on my own Windows 10.

<img width="924" alt="screen shot 2017-03-11 at 16 34 56" src="https://cloud.githubusercontent.com/assets/125426/23824540/bfd510ca-0678-11e7-9dd5-10dfe6b6c29b.png">

**Comment by obambrough** [comment]
Great - thanks for fixing it so quickly.

**Comment by obambrough** [comment]
@calmh Just tested with the 0.14.26 build and this issue is fixed (i.e. no 'privilege not held' error) but now the folder remains out of sync with a new error ('symlinks not supported'.) Previously there was an advanced setting to 'ignore symlinks' that disappeared in 0.14.24 which I think is still necessary, otherwise you end up with this:

![combined out of sync image](https://cloud.githubusercontent.com/assets/1922249/24681920/36e5234a-1965-11e7-81cf-6d5bfd679462.png)

**Comment by calmh** [comment]
I think the thinking was that symlinks are not special enough, in general, to warrant a toggle. Windows is the special case here though, when it is syncing against something non-Windows and also there are symlinks in that folder. The correct solution there is to ignore them by name using the regular ignore process.

**Comment by obambrough** [comment]
@calmh Thanks, I'll implement some naming scheme for symlinks so I can add a single ignore rule.

**Comment by lawrence-dol** [comment]
> The correct solution there is to ignore them by name using the regular ignore process.

Actually, arguably, the correct solution is to warn the user, give them the opportunity to ignore, and also instruct them that the special local security policy setting, "Security Settings / Local Policies / Security Options / Create symbolic links" is needed for non-Admins to manage symlinks.

**Comment by obambrough** [comment]
@calmh I implemented an ignore pattern for symlinks and it worked, my remote machine is no longer 'Out of Sync'. But there is one odd thing. The local machine that contains all the symlinks states a value of 7732 files under the 'Local State' in the UI. If I ignore a regular file that number will adjust, but when I added the ignore pattern for the symlinks that number doesn't change (I thought it'd go down by 97.) So the 'Local State' on the machine isn't counting the symlinks (whether there's an ignore pattern for them or not) but it obviously is sending the names of the symlinks (when there isn't an ignore pattern) to the remote machine (as the remote machine was previously complaining about the symlinks.) So until I was able to access the remote machine I didn't know if the ignore pattern for the symlinks was working.
